pkg/apis/serving/v1beta1/inference_service_defaults.go

```diff
@@ -38,6 +38,10 @@ var (
 	}
 	// logger for the mutating webhook.
 	mutatorLogger = logf.Log.WithName("inferenceservice-v1beta1-mutating-webhook")
+
+	// deprecatedFieldName is used for checking whether the deprecated predictor old schema is used or not.
+	// If it is then a warning is thrown by the validating webhook.
+	deprecatedFieldName string
```

Comment:
@yuzisun Here, I am using this package level variable to find if the deprecated field is used on the validating webhook side. By default, the defaulting webhook is called before the validating webhook. It is not possible to detect if the user is using the deprecated field in the validating webhook because we are converting the old schema to new schema in the defaulting webhook. 

We can change order in which the webhook is called. But It might be a breaking change so it will not be the best solution. If you have any better solution please, let me know.